File: Problem2.py

def fib(limit):
    a, b = 1, 0

    while a < limit:
        a, b = a+b, a
        yield a


if __name__ == '__main__':
    even_list = (x for x in fib(4000000) if x%2 == 0)
    print(sum(even_list))

File: Problem2.go

package main

import "fmt"

func main() {
    limit := 4000000
    sum := 0
    a := 1
    b := 0
    
    for a < limit {
        a, b = a + b, a

        if a % 2 == 0 {
            sum = sum + a 
        }
    }

    fmt.Println("Sum:", sum)
}
